[KOGITO-7646] Add event filter usage example to expression guide (#86)
kiegroup/kogito-docs: serverlessworkflow/modules/ROOT/pages/core/understanding-jq-expressions.adoc

* link:{spec_doc_url}#event-data-filters[Event data filters]: Similar to the action data filters, but apply to the events instead of actions.
* link:{spec_doc_url}#state-data-filters[State data filters]: Define the workflow model to the JSON object, which is returned by the expression and discards an existing property.

> === State and Action data filter usage

Reviewer: ```suggestion
State and Action data filter example::
```

> You can see link:{kogito_sw_examples_url}/serverless-workflow-expression-quarkus[`serverless-workflow-expression-quarkus`], in which actions and state data filters are used.

Reviewer: ```suggestion
You can see link:{kogito_sw_examples_url}/serverless-workflow-expression-quarkus[`serverless-workflow-expression-quarkus`] example application, in which actions and state data filters are used.
```

In the previous example, the workflow model is changed by the `input` property of the filter, which means that the output model is updated before the state is executed. As a final result, the output model consists of a `result` property, containing the maximum value of `x`.

> === Event data filter usage

Reviewer: ```suggestion
Event data filter example::
```

> You can find an example of event data filtering in the link:{kogito_sw_examples_url}/serverless-workflow-callback-quarkus[callback workflow application]. 

Reviewer: ```suggestion
You can find an example of event data filtering in the link:{kogito_sw_examples_url}/serverless-workflow-callback-quarkus[`serverless-workflow-callback-quarkus`] example application. 
```

> [source,json]

Reviewer: ```suggestion
.Example event filter
[source,json]
```

----
  "eventRef": "waitEvent",
  "eventDataFilter": {
     "data": ".result",
     "toStateData": ".move"
   }
----

> This event filter copies the content of Cloud Event data `result` field into Workflow model `move` field. 

Reviewer: ```suggestion
The previous example of the event filter copies the content of Cloud Event data `result` field into the workflow model `move` field. 
```

Author: why "the previous example of the" is better than "this"?
